src/argilla/server/apis/v1/handlers/datasets.py

```diff
@@ -59,6 +69,16 @@ def _get_dataset(db: Session, dataset_id: UUID):
     return dataset
 
 
+def _merge_search_records(search_responses: "SearchResponses", records: List[Records]) -> List[SearchRecord]:
+    search_records = []
+    for response in search_responses.items:
+        record = next((r for r in records if r.id == response.record_id), None)
+        if record:
+            search_records.append(SearchRecord(record=record, query_score=response.score))
+            records.remove(record)
+    return search_records
```

Comment:
maybe this function would be better off in another file, but I'm not sure where to put it, since it is an auxiliary function for the endpoint